### ISSUE TITLE
crosstab() pivot is not working

### ISSUE THREAD

**Issue by marcoruggine** [metabase/metabase#4046]
### Bug Report
- Your browser and the version: Chrome 55.0.2883.95 (64-bit)
- Your operating system: MacOs 10.12.2 
- Your databases: Postgres
- Metabase version: 0.21.1
- Metabase hosting environment: AWS
- Metabase internal database: Postgres

Hi, I've set up a query that runs correctly on our Postgres DB but when I paste it into MetaBase's SQL editor it pivots data in a completely wrong way.

Our data structure is fairly easy: we have a column for regions, one for months and another one with earnings that I'm summing up. Using crosstab() function, it should show regions in the first column, months as column headers and the overall earnings in the inner cells. Everything ordered by the highest earnings in a chosen month.

Unfortunately, in MetaBase instead of having months as column headers it plots random values, and so does in the inner cells.

I know It would be easier to use MB pivot option and a much simpler query, but then again I can't order by DESC/ASC in the chosen column.

Anyone has any idea on how to make it work? Thanks!

```sql
SELECT * FROM crosstab('
	SELECT regione::text, to_char(mese, ''YYYY/MM''), sum(incasso_puc)::numeric as incasso 
	FROM view_regional_history 
	WHERE mese between ''2016-01-01'' and ''2017-01-01''
	GROUP BY regione, mese
	ORDER BY regione DESC
	',
	'
	SELECT DISTINCT to_char(mese, ''YYYY/MM'') 
	FROM view_regional_history
	WHERE mese between ''2016-01-01'' and ''2017-01-01''
	'
)
AS ct(regione text, 
	"2016/01" numeric,
	"2016/02" numeric,
	"2016/03" numeric,
	"2016/04" numeric,
	"2016/05" numeric,
	"2016/06" numeric,
	"2016/07" numeric,
	"2016/08" numeric,
	"2016/09" numeric,
	"2016/10" numeric,
	"2016/11" numeric,
	"2016/12" numeric,
	"2017/01" numeric
	)
ORDER BY "2016/12" DESC NULLS LAST;
```

Here is the result from this query on our DB:
<img width="1160" alt="schermata 2017-01-03 alle 16 17 11" src="https://cloud.githubusercontent.com/assets/20041428/21612828/f28a24c2-d1d2-11e6-931c-ac2df94a146b.png">

Here is the result in MetaBase:
<img width="835" alt="schermata 2017-01-03 alle 16 17 21" src="https://cloud.githubusercontent.com/assets/20041428/21612840/ff8dd11e-d1d2-11e6-967d-683917aaf3ed.png">

**Comment by salsakran** [comment]
what type are `min_date` and `max_date`? Are you linking them to a field? If so, then you need to make sure the fully qualiified name eg (`tablename.fieldname`) amkes sense in each place a template variable is used.

Other than that I suggest taking pieces of that query that work, and let us know what exactly trips an error.

**Comment by salsakran** [comment]
also, regarding the "pivot" setting, that is meant to take an un-crosstab'd 2d grouping and transform it to a pivot table, so it's not surprising that leaving it enabled did something weird with an pre-pivoted table.

**Comment by marcoruggine** [comment]
Hi @salsakran thanks for getting in touch. 
I've figured out that the error I was getting was referred to the ' ' right before my variables.
I've tried `WHERE mese between {{min_date}} and {{max_date}}` and now is working properly, thanks for the help!

If I may suggest, it would be really helpful if Metabase could point at the exact position in the code where the error is, instead of saying "position:#", so to be able to spot quickly the typo/error. For example, I also use Postico and it does this:
```
ERROR:  syntax error at or near "YYYY"
LINE 1: SELECT regione::text, to_char(mese, ''YYYY/MM''), round(sum(...
                                              ^
```
